WordPress/src/main/java/org/wordpress/android/ui/plans/PlanUpdateService.java

```diff
@@ -0,0 +1,173 @@
+        Intent intent = new Intent(context, PlanUpdateService.class);
+        intent.putExtra(ARG_LOCAL_BLOG_ID, localTableBlogId);
+        context.startService(intent);
+    }
+
+    @Override
+    public IBinder onBind(Intent intent) {
+        return null;
+    }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        AppLog.i(AppLog.T.PLANS, "plan update service > created");
+    }
+
+    @Override
+    public void onDestroy() {
+        AppLog.i(AppLog.T.PLANS, "plan update service > destroyed");
+        super.onDestroy();
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        mLocalBlogId = intent.getIntExtra(ARG_LOCAL_BLOG_ID, 0);
+        downloadGlobalPlans();
+
+        return START_NOT_STICKY;
+    }
+
+    /*
+     * called when plan data has been successfully updated
+     */
+    private void notifySuccess(@NonNull List<SitePlan> plans) {
+        EventBus.getDefault().post(new PlanEvents.PlansUpdated(plans));
+    }
+
+    /*
+     * called when updating plan data fails
+     */
+    private void notifyFailure() {
+        EventBus.getDefault().post(new PlanEvents.PlansUpdateFailed());
+    }
+
+    /*
+     * Step 1: download global plans
+     */
+    private void downloadGlobalPlans() {
+        WordPress.getRestClientUtilsV1_3().get("plans/", WordPress.getRestLocaleParams(), null, new RestRequest.Listener() {
+            @Override
+            public void onResponse(JSONObject response) {
+                if (response != null) {
+                    AppLog.d(AppLog.T.PLANS, response.toString());
+                    // Store the response into App Prefs
+                    AppPrefs.setGlobalPlans(response.toString());
+                    // proceed to step 2
+                    downloadPlanFeatures();
+                } else {
+                    AppLog.w(AppLog.T.PLANS, "Empty response downloading global Plans");
+                    notifyFailure();
+                }
+            }
+        }, new RestRequest.ErrorListener() {
+            @Override
+            public void onErrorResponse(VolleyError volleyError) {
+                AppLog.e(AppLog.T.PLANS, "Error loading plans", volleyError);
+                notifyFailure();
+            }
+        });
+    }
+
+    /*
+     * Step 2: download plan features
+     */
+    private void downloadPlanFeatures() {
+        WordPress.getRestClientUtilsV1_2().get("plans/features/", WordPress.getRestLocaleParams(), null, new RestRequest.Listener() {
+            @Override
+            public void onResponse(JSONObject response) {
+                if (response != null) {
+                    AppLog.d(AppLog.T.PLANS, response.toString());
+                    // Store the response into App Prefs
+                    AppPrefs.setGlobalPlansFeatures(response.toString());
+                    // proceed to step 3
+                    downloadAvailablePlansForSite();
```

Comment:
Same of above. We can use Volley multithreading ability, and download all of them in parallel. It should boost the performance of this service.